### DISCUSSION TITLE
COG - Cloud Access S3/Swift/..

### DISCUSSION THREAD

**Discussion by EulenspiegelTill** [opengisch/QField#4412] [Q&A]
Is there the possiblity to access a CloudOptimized Geotiff somewhere in an Object storage, with authentification.

I tried arrount with COG in the cloud a bit. A Layer without auth. works in QField on my mac desktop, but on the Android tablet the error (picture) comes. So my first intention was to ask here what is possible and implemented in QField?

![230701_COG_QField](https://github.com/opengisch/QField/assets/23494522/e40678f3-e2bf-4710-8b11-e5bae22ca228)

**Comment by nirvn** [comment]
@EulenspiegelTill , we fixed an issue on Android which unlocked this very functionality. You can confirm it works using an APK provided here: https://github.com/opengisch/QField/pull/4415

**Reply by EulenspiegelTill** [reply]
Work's perfect! I will try arround a bit with compression and see whats working smooth. Also try out authentification (enviroment variables 🙃 ) on QGiS side with gdal, and will be back at you with that. but need some spare time first to try arround.

Thank's!

**Comment by SilvioMueller** [comment]
Since the newest version (3.4.7 - Ebo) I face this problem again. I can not access a public s3 COG from within QField but I have no problem accessing it with QGis. 

On another note, is there a way to set AWS credentidals in QField?

**Reply by nirvn** [reply]
@SilvioMueller , could you attach a sample project here with that public S3 cog layer? It'll be hard to look into this possible regression otherwise.

The latest version is now 3.5.0, have you tested there?

**Reply by SilvioMueller** [reply]
Thanks for looking into this.
I tried it in this project: silviom/AWS_Test2/. One important point I forgot to add, the COG is only accessible for some whitelisted IP's. I am trying to avoid to have my AWS objects completely public. It seems the problem is with QGIS Cloud:

```
08:46:27.882 QGSMSGLOG WARNING  Layer "Basel_4_2" cannot be packaged and will be removed because "PackagePreventionReason.INVALID".
08:46:27.882 libqfieldsync WARNING  Layer "Basel_4_2" cannot be packaged and will be removed because "PackagePreventionReason.INVALID".
```

**Reply by nirvn** [reply]
@SilvioMueller , ah, so, if you import your project into QField through other means (USB cable, import zipped project URL, etc.), does it work?

**Reply by suricactus** [reply]
@SilvioMueller thank you for these two key lines from the log! This is a `libqfieldsync` issue, tbh it was never tested directly with Object storage provider.

**Reply by SilvioMueller** [reply]
Ah I see! just for completeness, i exported the project directly to QField by plugging the Tablet. This also did not work
